airflow/jobs/scheduler_job_runner.py

```diff
@@ -1644,13 +1644,10 @@ def adopt_or_reset_orphaned_tasks(self, session: Session = NEW_SESSION) -> int:
 
                     query = (
                         select(TI)
+                        .options(lazyload("dag_run"))  # avoids double join to dag_run
                         .where(TI.state.in_(State.adoptable_states))
-                        # outerjoin is because we didn't use to have queued_by_job
-                        # set, so we need to pick up anything pre upgrade. This (and the
-                        # "or queued_by_job_id IS NONE") can go as soon as scheduler HA is
-                        # released.
```

Comment:
So I assume this happened? A note in the PR would be useful for back reference.

Reply:
good idear. yeah i'll add. re whether "this" happened, AFAICT "this" was airflow 2.0